consensus/src/messages/handlers.rs

```diff
@@ -154,10 +159,17 @@ impl<N: Network> Handle<N, Arc<RwLock<Blockchain>>> for RequestBatchSet {
 impl<N: Network> Handle<N, Arc<RwLock<Blockchain>>> for RequestHistoryChunk {
     fn handle(
         &self,
-        _peer_id: N::PeerId,
+        peer_id: N::PeerId,
         blockchain: &Arc<RwLock<Blockchain>>,
     ) -> Result<HistoryChunk, HistoryChunkError> {
-        if let Some(chunk) = blockchain.read().history_store.prove_chunk(
+        let blockchain = blockchain.read();
+
+        if blockchain.history_store.is_light() {
+            log::warn!(peer=%peer_id,"Request history chunk is not supported by the light history store");
+            return Err(HistoryChunkError::Other);
+        }
+
```

Comment:
I think it would be better if we didn't even listen for these requests in the light client. Instead of erroring out once we get a request.